project/ntut-xuan/README.rst

```diff
@@ -0,0 +1,198 @@
+PolygonDust
+===========
+
+Description
+-----------
+
+PolygonDust is an application to measure the area size of polygon.
+
+“Dust” means we trying to shred the image into lot of pieces and measure
+polygon size by these pieces.
+
+.. image:: https://media.discordapp.net/attachments/950048467294760990/1284862080050135071/image.png?ex=66e82c8f&is=66e6db0f&hm=8f52c071f46be06d7d6d014e8551087f495e2cf1332a784079c4805429e7aea3&=&format=webp&quality=lossless&width=2880&height=848
+
+Problem to Solve
+----------------
+
+In size measurement of ploygon, we have two methods can do it.
+
+-  Pixel-Method: Shred the image to lot of rectangles and calculate the
+   size by these rectangles.
+-  Formula-Method: Calculate the polygon by formula.
+   e.g. :math:`W \times H` for rectangle of :math:`r^2 \times \pi` for
+   circle.
+
+PolygonDust is used to solve size calculation by Pixel-Method. In
```

Comment:
Pixel-based algorithms for calculating area are not accurate.  If you go with the pixel-based algorithms, you should analyze the error, which requires you to develop a geometry ("formula") based algorithm.  Thus, a geometry-based code is needed anyway.

Reply:
I though may have Irregular Polygon in the image, and it's may hard to calculate the size area of polygon. Or should I don't consider Irregular Polygon case?

Reply:
By polygon, we usually mean [a geometry-based description using a node list](https://live.boost.org/doc/libs/1_86_0/libs/geometry/doc/html/geometry/reference/models/model_polygon.html) and it will be unusual if we do not handle arbitrary ones.  I am not sure what do you mean by a regular polygon.  Maybe you meant a rectangle or triangle?  They are uninteresting.

It is not clear to me how do you define a polygon in an image.  If you really want to handle a polygon in an image, you need to define it.  I think it is a very challenging problem and I am not aware of a solution.

Reply:
![image](https://github.com/user-attachments/assets/3aa8dfb0-5ab0-408c-bb44-2e29ae17793e)

These is irregular polygon I found.

Reply:
After I think about it, it may make sence if there are some third-party library to calculate the polygon area by geometry ("formula") based algorithm or I just can take a look about how to do it with source code. I'll consider the project loading and think about is it possible to working on it by myself or just using third-party library.

Reply:
After I "After I think about it", it may easy to calculate the polygon area by Shoelace formula with arbitrary polygon. I will try it.

Reply:
> After I "After I think about it", it may easy to calculate the polygon area by Shoelace formula with arbitrary polygon. I will try it.

Please include it in the proposal.  Also include how do you plan to handle holes.